extensions/mailer/runtime/src/main/java/io/quarkus/mailer/Attachment.java

```diff
@@ -0,0 +1,181 @@
+package io.quarkus.mailer;
+
+import java.io.File;
+
+/**
+ * Defines an attachment.
+ * <p>
+ * Instances of this class are not thread-safe.
+ */
+public class Attachment {
+
+    /**
+     * Disposition for inline attachments.
+     */
+    public static final String DISPOSITION_INLINE = "inline";
+
+    /**
+     * Disposition for attachments.
+     */
+    public static final String DISPOSITION_ATTACHMENT = "attachment";
+
+    private String name;
+    private File file;
+    private String description;
+    private String disposition;
+    private byte[] data;
```

Comment:
Shouldn't we make this stream-oriented somehow?

Reply:
Done, you can now pass a `Publisher<Byte>`.